lib/Dialect/TTIR/Transforms/Passes.cpp

```diff
@@ -614,14 +614,25 @@ createToLayoutOp(PatternRewriter &rewriter, Location loc, Value input,
   auto desiredLayout =
       rewriter.getAttr<LayoutAttr>(ty, desiredMemorySpace, currLayout.getGrid(),
                                    desiredElementType, desiredMemLayout);
+
+  ::mlir::TypedValue<::mlir::RankedTensorType> existingOpType = nullptr;
+  tensor::EmptyOp existingEmpty = input.getDefiningOp<tensor::EmptyOp>();
+  ttir::ConstantOp existingConstant = input.getDefiningOp<ttir::ConstantOp>();
+  if (existingEmpty) {
+    existingOpType = existingEmpty.getResult();
+  } else if (existingConstant) {
+    existingOpType = existingConstant.getResult();
+  }
```

Comment:
Can you please put an assert for the else {} case e.g. llvm_unreachable("message")

Reply:
Else is covered by the code below, in case of EmptyOp or ConstantOp we get the `existingOpType` otherwise it is null, and then afterwards if it isn't null we change it by inserting layout attr, otherwise we create new `to_layout` op.